app/controllers/items_controller.rb

```diff
@@ -536,10 +537,21 @@ def schema_validation
   end
 
   def refresh_metadata
+    if @object.catkey.blank?
+      render status: :forbidden, plain: 'object must have catkey to refresh descMetadata'
+      return
+    end
+
     @object.build_datastream('descMetadata', true)
     @object.descMetadata.content = @object.descMetadata.ng_xml.to_s
-    @object.descMetadata.save
-    render :status => :ok, :plain => 'Refreshed.'
```

Comment:
most of the operations in this controller use an `after_action` to handle the saving on the object as a whole.  so i moved responsibility out of the controller method and to that `after_action` (the first commit of this PR).  that `after_action` saves the object as a whole, as opposed to saving on the datastream specifically, but testing indicates that things still work fine with saving on the object as a whole.